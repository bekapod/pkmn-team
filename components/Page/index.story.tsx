import { storiesOf } from "@storybook/react";
import React from "react";
import Page from ".";

storiesOf("Page", module).add(
  "with content",
  (): JSX.Element => (
    <Page title="My Page">
      <div>Page Content</div>
    </Page>
  )
);
