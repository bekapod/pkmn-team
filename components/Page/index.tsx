import { ApolloError } from "apollo-client";
import React, { Fragment, ReactNode, ReactNodeArray } from "react";
import ErrorMessage from "../ErrorMessage";
import FullWidthContainer from "../FullWidthContainer";
import Heading from "../Heading";
import LoadingIcon from "../LoadingIcon";

interface IProps {
  title: string;
  loading: boolean;
  error?: ApolloError;
  children: ReactNode | ReactNodeArray;
}

const Page = ({ title, loading, error, children }: IProps) => (
  <Fragment>
    <Heading>{title}</Heading>
    <FullWidthContainer role="alert" aria-live="assertive">
      {loading ? <LoadingIcon /> : null}
      {!loading && error ? (
        <ErrorMessage isBig={true}>{error.message}</ErrorMessage>
      ) : null}
      {!loading && !error ? children : null}
    </FullWidthContainer>
  </Fragment>
);

export default Page;
