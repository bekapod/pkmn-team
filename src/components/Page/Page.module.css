.container {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
}

.heading {
  flex: 0;
}

.content {
  position: relative;
  flex: 1;
}
