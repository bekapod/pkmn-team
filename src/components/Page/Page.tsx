import type { ComponentPropsWithRef, FunctionComponent } from 'react';
import classNames from 'classnames';
import Head from 'next/head';
import { Header } from '../Header';

export type PageProps = ComponentPropsWithRef<'div'> & {
  title: string;
  metaTitle: string;
};

export const Page: FunctionComponent<PageProps> = ({
  className,
  title,
  metaTitle,
  children,
  ...props
}) => (
  <>
    <Head>
      <title>{metaTitle}</title>
      <link rel="icon" href="/favicon.ico" />
      <link rel="stylesheet" href="https://use.typekit.net/jdl7nve.css" />
    </Head>
    <div
      className={classNames('flex', 'flex-col', 'min-h-screen', className)}
      {...props}
    >
      <Header
        className={classNames('flex-grow-0', 'flex-shrink-0')}
        title={title}
      />
      <main className={classNames('relative', 'flex-grow-1', 'flex-shrink-1')}>
        {children}
      </main>
    </div>
  </>
);
