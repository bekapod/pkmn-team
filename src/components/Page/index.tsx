import { ApolloError } from "apollo-client";
import React, { Fragment, ReactNode, ReactNodeArray } from "react";
import ErrorMessage from "../ErrorMessage";
import Heading from "../Heading";
import LoadingIcon from "../LoadingIcon";
import SectionContainer from "../SectionContainer";

interface IProps {
  title: string;
  loading: boolean;
  error?: ApolloError;
  children: ReactNode | ReactNodeArray;
}

const Page = ({ title, loading, error, children }: IProps) => (
  <Fragment>
    <Heading>{title}</Heading>
    <SectionContainer role="alert" aria-live="assertive">
      {loading ? <LoadingIcon /> : null}
      {!loading && error ? (
        <ErrorMessage isBig={true}>{error.message}</ErrorMessage>
      ) : null}
      {!loading && !error ? children : null}
    </SectionContainer>
  </Fragment>
);

export default Page;
