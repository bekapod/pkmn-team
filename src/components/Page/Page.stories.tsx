import { Meta, Story } from '@storybook/react/types-6-0';
import { Page, PageProps } from './Page';
import { FullWidthContainer } from '../FullWidthContainer';

export default {
  title: 'Components/Page',
  component: Page,
  args: {
    title: 'Some heading',
    metaTitle: 'Some title'
  }
} as Meta<PageProps>;

export const page: Story<PageProps> = args => (
  <Page {...args}>
    <FullWidthContainer>A section of content</FullWidthContainer>
  </Page>
);
