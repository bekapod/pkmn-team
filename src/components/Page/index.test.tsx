import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import Page from ".";

describe("<Page />", () => {
  it("renders regular content", () => {
    const tree = renderer.create(
      <Page title="My Page" loading={false}>
        <div>Test Content</div>
      </Page>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders loading icon", () => {
    const tree = renderer.create(
      <Page title="My Page" loading={true}>
        <div>Test Content</div>
      </Page>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders error message", () => {
    const tree = renderer.create(
      <Page
        title="My Page"
        loading={false}
        error={{ message: "Error message." } as any}
      >
        <div>Test Content</div>
      </Page>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
