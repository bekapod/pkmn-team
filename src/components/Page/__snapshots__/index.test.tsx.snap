// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders error message 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Page
  </h1>,
  <div
    aria-live="assertive"
    className="SectionContainer-xjlkvu-0 jZbaLj"
    role="alert"
  >
    <div
      className="ErrorMessage__StyledError-sc-11kho77-0 DzHRJ"
      role="alert"
    >
      Error message.
    </div>
  </div>,
]
`;

exports[`<Page /> renders loading icon 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Page
  </h1>,
  <div
    aria-live="assertive"
    className="SectionContainer-xjlkvu-0 jZbaLj"
    role="alert"
  >
    <div
      className="LoadingIcon__Styled-sc-1njdtjv-0 gEufFU"
    >
      Loading
    </div>
  </div>,
]
`;

exports[`<Page /> renders regular content 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    My Page
  </h1>,
  <div
    aria-live="assertive"
    className="SectionContainer-xjlkvu-0 jZbaLj"
    role="alert"
  >
    <div>
      Test Content
    </div>
  </div>,
]
`;
